File: Bubble Sort/dart/bubble_sort.dart
import 'dart:io';

class BubbleSort {
  List? _arr;

  BubbleSort({List? arr}) {
    _arr = arr;
  }

  void sort({List? arr}) {
    if (arr != null) _arr = arr;

    for (int i = 0; i < _arr!.length - 1; i++) {
      for (int j = 0; j < _arr!.length - i - 1; j++) {
        if (_arr![j] > _arr![j + 1]) {
          _swap(j, j + 1);
        }
      }
    }
  }

  void _swap(int j, int i) {
    int temp = _arr![j];
    _arr![j] = _arr![i];
    _arr![i] = temp;
  }

  void display() {
    if (_arr != null) {
      stdout.write("[");
      for (int i = 0; i < _arr!.length; i++) {
        stdout.write('${_arr![i]}${i < _arr!.length - 1 ? ", " : ""}');
      }
      print("]");
    }
  }
}

File: Bubble Sort/java/src/BubbleSort.java
public class BubbleSort {
    private int[] arr;

    BubbleSort() {
    }

    BubbleSort(int[] arr) {
        this.arr = arr;
    }

    void sort() {
        for (int i = 0; arr.length - 1 > i; i++) {

            for (int j = 0; arr.length - i - 1 > j; j++) {
                if (arr[j] > arr[j + 1]) {
                    swap(j, j + 1);
                }
            }
        }
    }

    void sort(int[] arr) {
        this.arr = arr;
        sort();
    }

    private void swap(int j, int i) {
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }

    void display() {
        if (arr != null) {

            System.out.print("[");
            for (int i = 0; i < arr.length; i++) {
                System.out.print(arr[i] + ((i < arr.length - 1) ? ", " : ""));
            }
            System.out.println("]");
        }
    }

}
